core/templates/dev/head/pages/exploration_player/SolutionButtonDirective.js

```diff
@@ -26,6 +26,11 @@ oppia.directive('solutionButton', [
       },
       templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
         '/pages/exploration_player/' +
-        'solution_button_directive.html')
+        'solution_button_directive.html'),
+      controller: [
+        '$scope', 'WindowDimensionsService',
+        function($scope, WindowDimensionsService) {
+          $scope.isWindowNarrow = WindowDimensionsService.isWindowNarrow();
```

Comment:
Ditto -- can we leave the buttons as they are and do the ng-if in the parent container instead?